vertx-web/src/main/java/io/vertx/ext/web/handler/impl/SessionHandlerImpl.java

```diff
@@ -155,13 +160,13 @@ private void setCookieProperties(Cookie cookie) {
     }
   }
 
-  private SessionHandler flush(RoutingContext context, boolean skipCrc, Handler<AsyncResult<Void>> handler) {
+  private SessionHandler flush(RoutingContext context, boolean skipCrc, boolean ignoreStatus, Handler<AsyncResult<Void>> handler) {
```

Comment:
I think here we should rework the method to return a `Future` and have the Handler<AsyncResult> method delegate to the Future method, as modernization of the code base to use futures.

Reply:
Hmm, is there a good example of how I should model that? Maybe that would be best as a follow up refactoring PR since we do at least have a `Future` interface wrapper for `flush` already.

Reply:
I agree with @vietj however this is an already existing method. Shouldn't we just create an overall task for the whole module to "futurize" the internals? I mean Most of the internals are still using callbacks, so a full thorough (but unrelated task) should pick this one too. WDYT?

Reply:
Yeah, I think doing that separately from this PR would be best.